src/software/ai/evaluation/calc_best_shot.cpp

```diff
@@ -1,20 +1,137 @@
 #include "software/ai/evaluation/calc_best_shot.h"
 
-#include "software/ai/evaluation/calc_best_shot_impl.h"
-#include "software/geom/algorithms/acute_angle.h"
-#include "software/geom/algorithms/multiple_segments.h"
-#include "software/geom/algorithms/projection.h"
-
 std::optional<Shot> calcBestShotOnGoal(const Segment &goal_post, const Point &shot_origin,
-                                       const std::vector<Robot> &robot_obstacles)
+                                       const std::vector<Robot> &robot_obstacles,
+                                       TeamType team_type, double radius)
 {
-    // Use shot evaluation function to get the best Shot
-    std::vector<Circle> obs;
-    for (Robot robot : robot_obstacles)
+    if (team_type == TeamType::ENEMY)
+    {
+        Angle pos_post_angle = (goal_post.getStart() - shot_origin).orientation();
```

Comment:
there's a lot of duplicated code between the if and else statements that can probably be factored into a helper function

Reply:
It looks like there is, but because of the different angle mappings it'll be weird to factor stuff out

Reply:
yeah but this is a code maintainability and readability issue

Reply:
sort of done? let me know what you think